PrepPy/train_valid_test_split.py

```diff
@@ -0,0 +1,139 @@
+      
+    valid_size: float or None, (default=None)
+        If float, a value between 0.0 and 1.0 to represent the proportion of the input dataset to
+          comprise the size of the validation subset
+        If None, the value will be set to 0.0
+      
+    train_size: float or None, (default=None)
+        If float, a value between 0.0 and 1.0 to represent the proportion of the input dataset to
+          comprise the size of the train subset
+        If None, the value will be set to the complement of the test_size and valid_size
+  
+    stratify: array-like or None (default=None)
+        If not None, splits categorical data in a stratified fashion preserving the same proportion of
+            classes in the train, valid and test sets, using this input as the class labels    
+      
+    random_state: integer, optional (default=None)
+        A value for the seed to be used by the random number generator
+        If None, the value will be set to 1
+      
+    shuffle: logical, optional (default=TRUE)
+        Indicate whether data is to be shuffled prior to splitting
+  
+    Returns
+    -------
+    splits: list, length = 3 * len(arrays)
+        List containing train, validation and test splits of the input data
+      
+    Examples
+    --------
+    >>> from PrepPy import PrepPy as pp
+    >>> X, y = np.arange(16).reshape((8, 2)), list(range(8))
+    >>> X
+    array([[0, 1],
+          [2, 3],
+          [4, 5],
+          [6, 7],
+          [8, 9],
+          [10, 11],
+          [12, 13],
+          [14, 15]])
+           
+    >>> list(y)
+    [0, 1, 2, 3, 4, 5, 6, 7]
+    
+    >>> X_train, X_valid, X_test, y_train, y_valid, y_test = pp.train_valid_test_split(X, y, test_size=0.25, valid_size=0.25, random_state=777)
+
+    >>> X_train
+    array([[4, 5],
+          [0, 1],
+          [6, 7],
+          [12, 13]])
+           
+    >>> y_train
+    [3, 0, 2, 5]
+
+    >>> X_valid
+    array([[2, 3],
+         [10, 11]])
+           
+    >>> y_valid
+    [1, 4]
+  
+    >>> X_test
+    array([[8, 9],
+         [14, 15]])
+           
+    >>> y_test
+    [7, 6]  
+  
+  
+    >>> pp.train_valid_test_split(X, test_size=2, shuffle=False)
+
+    >>> X_train
+    array([[2, 3],
+          [14, 15],
+          [6, 7],
+          [12, 13],
+          [4, 5],
+          [10, 11]])
+   
+    >>> X_test
+    array([[8, 9],
+          [0, 1]])        
+    """
+
+    # Check data input types and parameters
+    assert ((isinstance(X, pd.DataFrame)) or (isinstance(X, np.ndarray)) or (isinstance(X, pd.DataFrame))), "Please input a list, numpy array, scipy-sparse matrices or pandas dataframe"
+    assert ((isinstance(y, pd.DataFrame)) or (isinstance(X, np.ndarray)) or (isinstance(X, pd.DataFrame))), "Please input a list, numpy array, scipy-sparse matrices or pandas dataframe"
+
+    assert len(X) != 0, "Your input is empty"
+    assert len(y) != 0, "Your input is empty"
```

Comment:
Assert should be part of the separate test file. You may include exception

Reply:
I will move these to `train_valid_test_split_test.py`